rebasehelper/output_tool.py

```diff
@@ -19,216 +19,177 @@
 #
 # Authors: Petr Hracek <phracek@redhat.com>
 #          Tomas Hozza <thozza@redhat.com>
+from __future__ import print_function
 
-import sys
 import os
 import six
-import json
+import pkg_resources
 
-from rebasehelper.exceptions import RebaseHelperError
-from rebasehelper.logger import LoggerHelper, logger, logger_report
+from rebasehelper.logger import logger
 from rebasehelper.results_store import results_store
-
-output_tools = {}
-
-
-def register_output_tool(output_tool):
-    output_tools[output_tool.PRINT] = output_tool
-    return output_tool
+from colors import red, green, yellow
 
 
 class BaseOutputTool(object):
-
     """
-    Class used for testing and other future stuff, ...
-    Each method should overwrite method like run_check
+    Base class for OutputTools.
+    print_cli_summary must be overridden in order to produce different CLI output
     """
 
     DEFAULT = False
+    NAME = 'name'
 
     @classmethod
-    def match(cls, cmd=None, *args, **kwargs):
-        """Checks if tool name matches the desired one."""
-        raise NotImplementedError()
+    def get_report_path(cls, app):
+        return os.path.join(app.results_dir, 'report.' + cls.NAME)
```

Comment:
I don't like this very much. For example `text` output tool should definitely be named `text`, but the generated file I think should rather have standard `.txt` extension.